src/bindings/python/src/pyopenvino/graph/node.cpp

```diff
@@ -80,6 +83,53 @@ void regclass_graph_Node(py::module m) {
         }
         return "<" + type_name + ": '" + self.get_friendly_name() + "' (" + shapes_ss.str() + ")>";
     });
+
+    // This function replaces NodeFactory's mechanism previously getting attributes of Nodes.
+    // Python will call this method whenever requested attribute hasn't already been defined,
+    // so all other properties like shape or name are prioritized from bindings itself.
+    // TODO: is it possible to append these attributes to the instance of Node itself
+    //       directly from pybind without returning class itself everytime?
+    node.def("__getattr__", [](const std::shared_ptr<ov::Node>& self, const std::string& name) {
+        // TODO: is it possible to cache serializer and regex?
+        util::DictAttributeSerializer dict_serializer(self);
+
+        // Look if there is a "get/set_*" pattern at the start of a string.
+        // It means attribute was called as a function.
+        std::smatch regex_match;
+        std::regex look_for("get_|set_");
+
+        if (std::regex_search(name, regex_match, look_for, std::regex_constants::match_continuous)) {
+            // Strip prefix from the name and perform operation.
+            auto stripped_name = regex_match.suffix().str();
+            if (regex_match.str() == "get_") {
```

Comment:
Is it certain that this regex will work for all cases?

Reply:
As far as I know, yes. All previous attributes were passed as "get/set_name_of_the_attr", so only checking on the start of string should be performed, thus same matching is done here.